### PR TITLE
95 refactor unit tests

### LINKED ISSUE #95: Unit tests
- [ ] Remove too long tests in `test_dnn.py`
- [ ] Add unit tests to `jackknife.py`
- [ ] Add unit tests to `model.py`

### PR DESCRIPTION
**Pull request by rogerkuou**
fix #95

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi @SarahAlidoost, when you have time, could you please also review this one for me? I improved the unit tests to increase the coverage and speed.

File: motrainer/jackknife.py [hunk 1]
```diff
@@ -223,7 +223,9 @@ def export_best(self, model_name='best_optimized_model',
         metedata['input_list'] = self.input_list
         metedata['output_list'] = self.input_list
         metedata['best_year'] = int(self.best_year)
-        metedata['lat'] = float(self.gpi_data['lat'].iloc[0])
-        metedata['lon'] = float(self.gpi_data['lon'].iloc[0])
+        if 'lat' in self.gpi_data.keys():
+            metedata['lat'] = float(self.gpi_data['lat'].iloc[0])
+        if 'lon' in self.gpi_data.keys():
+            metedata['lon'] = float(self.gpi_data['lon'].iloc[0])
```
Comment (SarahAlidoost) [inline motrainer/jackknife.py:229]: I like that you implemented the check for coordinates. This part of the code is also changed in #97 with respects to the coordinate names that should be `latitude` and `longitude`. can you please merge the `generalization` into your branch and resolve the merge conflicts?

File: motrainer/util.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import os
 import sklearn.preprocessing
 import random
-from scipy.stats.stats import pearsonr, spearmanr
+from scipy.stats import pearsonr, spearmanr
 
 os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'  # Force tensorflow debug logging off
```
Comment (SarahAlidoost) [inline motrainer/util.py:8]: This file is also changed in #97, can you please merge the generalization into your branch? There, I changed the value of `os.environ['TF_CPP_MIN_LOG_LEVEL']` from '3' to '2'. :smile:

File: tests/test_dnn.py [hunk 3]
```diff
@@ -1,132 +1,113 @@
...
+EXPECTED_DIMS = set(
+    [
+        "learning_rate",
+        "num_dense_layers",
+        "num_input_nodes",
+        "num_dense_nodes",
+        "activation",
+        "batch_size",
+    ]
+)
```
Comment (SarahAlidoost) [inline tests/test_dnn.py:37]:
```suggestion
EXPECTED_DIMS = {
    "learning_rate",
    "num_dense_layers",
    "num_input_nodes",
    "num_dense_nodes",
    "activation",
    "batch_size",
}
```

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@rogerkuou nice work :+1:  please see my comments. One question: You mentioned that this pull request improves the speed of the tests. I am just curious about how you measured it. how much improvement in the speed is reached? :thinking:

**Comment by rogerkuou** [comment]
Hi @SarahAlidoost, thanks for the review!
I applied most of your comments. Apart from the one on checking `best_loss` and `history`, see my comments. Could you please give another look?

**Comment by SarahAlidoost** [comment]
> Hi @SarahAlidoost, thanks for the review! I applied most of your comments. Apart from the one on checking `best_loss` and `history`, see my comments. Could you please give another look?

Thanks for addressing the comments, feel free to merge :+1:

**Pull request merged**